### PR TITLE
Tests for utils func and e2e tests

### PR DESCRIPTION
**Pull request by Gochad**
Simple tests for functions:
1. `title_normalizer`
2. `jiwer_differ`
3. e2e command tests
4. test for dummy model funcs

### CODE REVIEW

File: apptests/test_e2e.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        json_file_path = os.path.join(target_directory, filename)
+        self.assertTrue(
+            os.path.exists(json_file_path), f"{json_file_path} does not exist."
+        )
+
+        with open(json_file_path, "r") as json_file:
+            try:
+                json.load(json_file)
+            except json.JSONDecodeError:
+                self.fail(f"{json_file_path} is not a valid JSON file.")
```
Comment (Kowalski1024) [inline apptests/test_e2e.py]: I think this should create a folder for the test plan in the apptests directory, and then both the new directory and the test plan should be deleted as part of the cleanup.

File: apptests/test_e2e.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        command = (
+            "python3 generators/youtube_generator.py 10 -o ./testplans -c 19 -l en"
+        )
+        subprocess.run(command, shell=True)
+        time_str = time.strftime("%Y%m%d-%H%M%S")
+        filename = f"Travel&Events_en_CAUQAQ_{time_str}.json"
+
+        command = (
+            f"python3 youtube_runner.py './testplans/{filename}' -st DummyTest -m tiny"
+        )
```
Comment (Kowalski1024) [inline apptests/test_e2e.py]: use an existing json file, such as `apptests/data/simpleTest.json`

File: apptests/test_e2e.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        self.assertEqual(output_data["failures"], 1, "Expected test failure not found.")
+        self.assertEqual(output_data["testsRun"], 2, "Unexpected number of tests run.")
+
+    def test_testplan_execute_proper_model(self):
+        command = (
+            "python3 generators/youtube_generator.py 10 -o ./testplans -c 19 -l en"
+        )
+        subprocess.run(command, shell=True)
+        time_str = time.strftime("%Y%m%d-%H%M%S")
+        filename = f"Travel&Events_en_CAUQAQ_{time_str}.json"
```
Comment (Kowalski1024) [inline apptests/test_e2e.py]: use an existing json file, such as `apptests/data/simpleTest.json`

File: apptests/test_e2e.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        self.assertEqual(output_data["failures"], 1, "Expected test failure not found.")
+        self.assertEqual(output_data["testsRun"], 2, "Unexpected number of tests run.")
+
+    def test_execute_with_missing_parameter(self):
+        command = (
+            "python3 generators/youtube_generator.py 10 -o ./testplans -c 19 -l en"
+        )
+        subprocess.run(command, shell=True)
+        time_str = time.strftime("%Y%m%d-%H%M%S")
+        filename = f"Travel&Events_en_CAUQAQ_{time_str}.json"
```
Comment (Kowalski1024) [inline apptests/test_e2e.py]: use an existing json file, such as `apptests/data/simpleTest.json`

File: apptests/test_e2e.py (hunk 1 above)
Comment (Gochad) [reply apptests/test_e2e.py]: Dir for test generated testsplan is `apptests/testplans` and this dir is removed after "good path" test

**Pull request merged**